packages/js/components/src/select-control/index.tsx

```
@@ -412,13 +417,16 @@ export class SelectControl extends Component< Props, State > {
 	getOptionsByQuery( options: Option[], query: string | null ) {
-		const { getSearchExpression, maxResults, onFilter } = this.props;
+		const { getSearchExpression, maxResults, onFilter, ignoreDiacritics } = this.props;
 		const filtered = [];
 
 		// Create a regular expression to filter the options.
-		const expression = getSearchExpression!(
-			escapeRegExp( query ? query.trim() : '' )
-		);
+		const baseQuery = query ? query.trim() : '';
+		const normalizedQuery = ignoreDiacritics
+			? baseQuery.normalize( 'NFD' ).replace( /[\u0300-\u036f]/g, '' )
```

Biome (1.9.4)

[error] A character class cannot match a character and a combining character.

A character and a combining character forms a new character. Replace the character class with an alternation.

(lint/suspicious/noMisleadingCharacterClass)